src/collective/cover/tiles/calendar.py

```diff
@@ -56,31 +61,57 @@ def _setup(self):
 
         self.monthName = PLMF(self._ts.month_msgid(month),
                               default=self._ts.month_english(month))
+        self._set_first_weekday()
+
+    def _set_first_weekday(self):
+        try:  # Plone 5.x
+            first_weekday = api.portal.get_registry_record('plone.first_weekday')
+        except InvalidParameterError:
+            try:  # plone.app.event
+                first_weekday = api.portal.get_registry_record('plone.app.event.first_weekday')
+            except InvalidParameterError:  # Plone 4.x
+                portal_calendar = api.portal.get_tool('portal_calendar')
+                first_weekday = getattr(portal_calendar, 'firstweekday', calendar.SUNDAY)
+        calendar.setfirstweekday(first_weekday)
 
     def accepted_ct(self):
         """Return an empty list as no content types are accepted."""
         return []
 
-    def getEventsForCalendar(self):
-        context = aq_inner(self.context)
-        year = self.year
-        month = self.month
-        portal_state = getMultiAdapter((self.context, self.request), name='plone_portal_state')
-        navigation_root_path = portal_state.navigation_root_path()
-        weeks = self.calendar.getEventsForCalendar(month, year, path=navigation_root_path)
-        for week in weeks:
-            for day in week:
-                daynumber = day['day']
-                if daynumber == 0:
-                    continue
-                day['is_today'] = self.isToday(daynumber)
-                if day['event']:
-                    cur_date = DateTime(year, month, daynumber)
-                    localized_date = [self._ts.ulocalized_time(cur_date, context=context, request=self.request)]
-                    day['eventstring'] = '\n'.join(localized_date + [
-                        ' {0}'.format(self.getEventString(e)) for e in day['eventslist']])
-                    day['date_string'] = '{0}-{1}-{2}'.format(year, month, daynumber)
+    def addEvents(self, day):
+        catalog = api.portal.get_tool('portal_catalog')
```

Comment:
@rodfersou the catalog call must be outside of the loop or you are going to call it up to 5 times instead of just one.